app/models/experiment.rb

```diff
@@ -331,23 +331,22 @@ def create_scatter_plot_csv_for(x_axis, y_axis)
     CSV.generate do |csv|
       csv << [x_axis, y_axis, 'simulation_run_ind']
 
-      simulation_runs.where({is_done: true, is_error: {'$exists' => false}}, {fields: %w(index values result arguments)}).each do |simulation_run|
+      simulation_runs.where({is_done: true, is_error: {'$exists' => false}}).each do |simulation_run|
         simulation_run_ind = simulation_run.index.to_s
-        simulation_input = Hash[simulation_run.arguments.split(',').zip(simulation_run.values.split(','))]
 
         x_axis_value = if simulation_run.result.include?(x_axis)
                          # this is a MoE
                          simulation_run.result[x_axis]
                        else
                          # this is an input parameter
-                         simulation_input[x_axis]
+                         simulation_run.input_parameters[x_axis]
```

Comment:
@dkrol - ze względu na to, że input_parameters w przypadku starych simulation runów tworzy hasha i w tej pętli jest to wywoływane 2x oraz sama pętla wywoływana jest tyle razy ile simulation_runów (czasami dużo) to proponuję podstawić wartość ``simulation_input = simulation_run.input_parameters`` wyżej.